crates/soft-confirmation-rule-enforcer/src/hooks.rs

```diff
@@ -104,17 +97,14 @@ where
         &self,
         soft_confirmation: &mut HookSoftConfirmationInfo,
         working_set: &mut WorkingSet<C>,
-    ) -> Result<(), ApplySoftConfirmationError> {
+    ) -> Result<(), SoftConfirmationError> {
         let current_timestamp = soft_confirmation.timestamp();
         let last_timestamp = self.last_timestamp.get(working_set).unwrap_or(0);
 
         if current_timestamp < last_timestamp {
-            return Err(
-                ApplySoftConfirmationError::CurrentTimestampIsNotGreaterThanPrev {
```

Comment:
If we don't care about error as enum maybe just return anyhow::Result<R, String>?

Reply:
https://github.com/chainwayxyz/citrea/issues/1048